src/gui/modalMenu.cpp

```diff
@@ -112,6 +120,71 @@ void GUIModalMenu::removeChildren()
 	}
 }
 
+// clang-format off
+bool GUIModalMenu::DoubleClickDetection(const SEvent &event)
+{
+	/* The following code is for capturing double-clicks of the mouse button
+	 * and translating the double-click into an EET_KEY_INPUT_EVENT event
+	 * -- which closes the form -- under some circumstances.
+	 *
+	 * There have been many github issues reporting this as a bug even though it
+	 * was an intended feature.  For this reason, remapping the double-click as
+	 * an ESC must be explicitly set when creating this class via the
+	 * /p remap_dbl_click parameter of the constructor.
+	 */
+
+	if (!m_remap_dbl_click)
+		return false;
+
+	if (event.MouseInput.Event == EMIE_LMOUSE_PRESSED_DOWN) {
+		m_doubleclickdetect[0].pos = m_doubleclickdetect[1].pos;
+		m_doubleclickdetect[0].time = m_doubleclickdetect[1].time;
+
+		m_doubleclickdetect[1].pos = m_pointer;
+		m_doubleclickdetect[1].time = porting::getTimeMs();
+	} else if (event.MouseInput.Event == EMIE_LMOUSE_LEFT_UP) {
+		u64 delta = porting::getDeltaMs(
+			m_doubleclickdetect[0].time, porting::getTimeMs());
+		if (delta > 400) {
+			return false;
+		}
```

Comment:
Possible code quality improvement.
> Do not unnecessarily use braces where a single statement will do.

From the Linux code style inherited rule. Just in case you'd like to tidy things a bit up in the meantime.